dm/dm/master/server.go

```diff
@@ -2145,9 +2146,61 @@ func (s *Server) GetCfg(ctx context.Context, req *pb.GetCfgRequest) (*pb.GetCfgR
 	if shouldRet {
 		return resp2, err2
 	}
-	// For the get-config command, you want to filter out fields that are not easily readable by humans,
-	// such as SSLXXBytes field in `Security` struct
+
+	formartTaskString := func(subCfgList []*config.SubTaskConfig) string {
+		sort.Slice(subCfgList, func(i, j int) bool {
+			return subCfgList[i].SourceID < subCfgList[j].SourceID
+		})
+		// For the get-config command, we want to filter out fields that are not easily readable by humans,
+		// such as SSLXXBytes field in `Security` struct
+		taskCfg := config.SubTaskConfigsToTaskConfig(subCfgList...)
+		taskCfg.TargetDB.Password = "******"
+		if taskCfg.TargetDB.Security != nil {
+			taskCfg.TargetDB.Security.ClearSSLBytesData()
+		}
+		return taskCfg.String()
+	}
 	switch req.Type {
+	case pb.CfgType_TaskTemplateType:
+		task, err := ha.GetOpenAPITaskTemplate(s.etcdClient, req.Name)
+		if err != nil {
+			resp2.Msg = err.Error()
+			// nolint:nilerr
+			return resp2, nil
+		}
+		if task == nil {
+			resp2.Msg = "task not found"
+			// nolint:nilerr
+			return resp2, nil
+		}
+		toDBCfg := config.GetTargetDBCfgFromOpenAPITask(task)
+		if adjustDBErr := adjustTargetDB(ctx, toDBCfg); adjustDBErr != nil {
+			if err != nil {
+				resp2.Msg = err.Error()
+				// nolint:nilerr
+				return resp2, nil
+			}
+		}
+		sourceCfgMap := make(map[string]*config.SourceConfig)
+		for _, cfg := range task.SourceConfig.SourceConf {
+			if sourceCfg := s.scheduler.GetSourceCfgByID(cfg.SourceName); sourceCfg != nil {
+				sourceCfgMap[cfg.SourceName] = sourceCfg
+			} else {
+				resp2.Msg = fmt.Sprintf("the source: %s of task not found", cfg.SourceName)
+				return resp2, nil
+			}
+		}
+		subTaskConfigList, err := config.OpenAPITaskToSubTaskConfigs(task, toDBCfg, sourceCfgMap)
```

Comment:
I prefer we change the return type of `OpenAPITaskToSubTaskConfigs` to []*SubTaskConfig

Reply:
add in https://github.com/pingcap/tiflow/pull/3991/commits/7be509f333f20075e9c167c61a02adb05926fa6e, and fix many typos (i opened spell checker for shell scripts now 😂)